app/models/course/discussion/post.rb

```diff
@@ -86,6 +86,16 @@ def cast_vote!(user, vote)
     end
   end
 
+  # Mark/unmark post as the correct answer.
+  def toggle_answer
```

Comment:
Can add specs for this and `update_resolve_status`?  Must check for the rollback condition.

Reply:
actually it's not supposed to fail, it add little value testing that branch, I'd like to skip it in this case, what you think? 😄

Reply:
Hmm, ok maybe minimally test the happy path.

Reply:
Yes, the other common path is tested already.